pontoon/base/models.py

```diff
@@ -373,7 +375,8 @@ def locale_checkout_path(self, locale):
         if not self.multi_locale:
             raise ValueError('Cannot get locale_checkout_path for non-multi-locale repos.')
 
-        return os.path.join(self.checkout_path, locale.code)
+        path_components = [self.project.checkout_path] + urlparse(self.url).path.split('/')
+        return os.path.join(*path_components).format(locale_code=locale.code)
```

Comment:
Ah, that makes sense. We might not actually support those correctly currently, although I'm not entirely sure. The issue is that all of our code about resource paths makes an assumption that everything below the source directory (which is named en-US, en, or templates) and everything below the locale directories is identical in structure. I _think_ that assumption holds currently, but I'm not 100%.

In any case, restoring it is the right move for this PR, so we're good here.